evaluator/perception_online_evaluator/src/perception_online_evaluator_node.cpp

```diff
@@ -132,50 +132,76 @@ void PerceptionOnlineEvaluatorNode::publishDebugMarker()
     tier4_autoware_utils::appendMarkerArray(added, &marker);
   };
 
-  const auto history_path_map = metrics_calculator_.getHistoryPathMap();
   const auto & p = parameters_->debug_marker_parameters;
 
-  for (const auto & [uuid, history_path] : history_path_map) {
-    {
-      const auto c = createColorFromString(uuid + "_raw");
-      if (p.show_history_path) {
-        add(createPointsMarkerArray(history_path.first, "history_path_" + uuid, 0, c.r, c.g, c.b));
+  // visualize history path
+  {
+    const auto history_path_map = metrics_calculator_.getHistoryPathMap();
+    int32_t history_path_first_id = 0;
+    int32_t smoothed_history_path_first_id = 0;
+    size_t i = 0;
+    for (auto it = history_path_map.begin(); it != history_path_map.end(); ++it) {
+      const auto & [uuid, history_path] = *it;
+      {
+        const auto c = createColorFromString(uuid + "_raw");
+        if (p.show_history_path) {
+          add(createPointsMarkerArray(history_path.first, "history_path", i, c.r, c.g, c.b));
+        }
+        if (p.show_history_path_arrows) {
+          add(createPosesMarkerArray(
+            history_path.first, "history_path_arrows", history_path_first_id, c.r, c.g, c.b, 0.1,
+            0.05, 0.05));
+          history_path_first_id += history_path.first.size();
+        }
       }
-      if (p.show_history_path_arrows) {
-        add(createPosesMarkerArray(
-          history_path.first, "history_path_arrows_" + uuid, c.r, c.g, c.b, 0.1, 0.05, 0.05));
+      {
+        const auto c = createColorFromString(uuid);
+        if (p.show_smoothed_history_path) {
+          add(createPointsMarkerArray(
+            history_path.second, "smoothed_history_path", i, c.r, c.g, c.b));
+        }
+        if (p.show_smoothed_history_path_arrows) {
+          add(createPosesMarkerArray(
+            history_path.second, "smoothed_history_path_arrows", smoothed_history_path_first_id,
+            c.r, c.g, c.b, 0.1, 0.05, 0.05));
+          smoothed_history_path_first_id += history_path.second.size();
+        }
       }
+      i++;
     }
-    {
+  }
+
+  // visualize predicted path of past objects
+  {
+    int32_t predicted_path_first_id = 0;
+    int32_t history_path_first_id = 0;
+    int32_t deviation_lines_first_id = 0;
+    size_t i = 0;
+    const auto object_data_map = metrics_calculator_.getDebugObjectData();
+    for (auto it = object_data_map.begin(); it != object_data_map.end(); ++it) {
```

Comment:
Can you change the format like following:

```
for (const auto & [uuid, histry_path] : history_path_map) {
...
}
```

Reply:
thanks, fixed in 6ea8a0faa80fdece84e892d988c97b04d4998d83